### DISCUSSION TITLE
Tuple with ellipsis

### DISCUSSION THREAD

**Discussion by AlfreG** [python/typing#993] [General]
Dear all,
I am puzzled about this, why prior assignment of type is giving me this error while explicit annotation worked?

test.py:5: error: Unexpected "..."
YY = tuple[int, ...]_

from typing import Tuple
YY = tuple[int, ...]

def it_works_fine(par: tuple[int, ...] = (1, 1,) ) -> bool:
    """."""
    return par == (1,1, )

def doesnt_work(par: YY = (1,1,) ) -> bool:
    return par == (1,1, )

def doesnt_work_neither(par: tuple[int, ...] = (1, 1,)  ) -> YY:
    return par

**Comment by srittau** [comment]
Which version of mypy are you using? Version prior to 0.930 (or 0.920?) had problems with tuples of the form `tuple[X, ...]`. Either upgrade mypy or use `typing.Tuple[X, ...]`.

**Reply by AlfreG** [reply]
I am using 0.910 indeed, thanks a lot!
